product_docs/docs/pgd/5/admin-tpa/upgrading_major_rolling.mdx

```diff
@@ -0,0 +1,537 @@
+---
+title: Performing a Postgres major version rolling upgrade on a PGD cluster built with TPA
+navTitle: Upgrade Postgres major version
+deepToC: true
+---
+
+## Overview
+
+You can perform a major version upgrade of Postgres on a Postgres Distributed (PGD) cluster deployed by Trusted Postgres Architect (TPA).
+
+The rolling upgrade process allows updating individual cluster nodes to a new major Postgres version while maintaining cluster availability and operational continuity. This approach minimizes downtime and ensures data integrity by allowing the rest of the cluster to remain operational as each node is upgraded sequentially. 
+
+We recommend upgrading your Postgres cluster to a newer major version to access improved features, performance enhancements, and security updates. These step-by-step instructions and a worked example help to provide a smooth and controlled upgrade process.
+
+### Plan the upgrade
+
+The cluster must be available throughout the process (that is, a *rolling* upgrade). To achieve continuous cluster availability, you must [perform a planned switchover](#performing-a-planned-switchover) of the [write leader](https://www.enterprisedb.com/docs/pgd/latest/terminology/#write-leader) node before upgrading it so that a write leader is always available. 
+
+You can see which node is the write leader by connecting to the PGD cluster (for example, using SSH) and running the `pgd show-groups` command:
```

Comment:
Missing - discussion about groups and subgroups. You need to bear in mind that you need to know what group/subgroup the nodes are in because its this group you are going to be rolling upgrading.
Also may need a note for when there are two regions... and no regions (you are upgrading nodes in the top-level group only). We can discuss.

Reply:
I am not following this 100%.  I could include a brief explanation of how groups and subgroups work in a PGD cluster. I thought the only relevant information using the show-groups command here was to learn which node is the write leader. We need to know which nodes belong to which group/subgroup? Could you explain this a bit more?

Reply:
If you have two subgroups, you'll have a write leader in each group. (Possibly three, not sure if you could configure the top-level with routing too).... 

So, you need to know what sub-group you are upgrading because it has its own write leader and that's what the switchover will change.

Reply:
Okay I will add a little something on this near the beginning on preparing/planning the upgrade.